library/common/http/client.cc

```diff
@@ -585,53 +584,8 @@ void Client::removeStream(envoy_stream_t stream_handle) {
 }
 
 namespace {
-
-const LowerCaseString ClusterHeader{"x-envoy-mobile-cluster"};
 const LowerCaseString H2UpstreamHeader{"x-envoy-mobile-upstream-protocol"};
```

Comment:
@goaway on Friday we discussed having:
1. :scheme dictate if the request is routed to non-tls http1.1
2. route specifically to h2 if the hostname is included.
3. route everything else to alpn.

Looking at the code that uses `x-envoy-mobile-upstream-protocol` it seems that we have the gRPC platform bindings force routing to h2. I can preserve that functionality. But do we want to allow for _any_ request to force routing via `x-envoy-mobile-upstream-protocol`?

Reply:
I think I might remove the protocol-based mapping (since it's incomplete anyways) and just leave in the cluster override header and use that for the gRPC case.